dask/dataframe/shuffle.py

```diff
@@ -103,12 +103,19 @@ def remove_nans(divisions):
     [1, 1, 2]
     >>> remove_nans((1, np.nan, 2))
     [1, 2, 2]
+    >>> remove_nans((1, 2, np.nan))
+    [1, 2, 2]
     """
     divisions = list(divisions)
+
     for i in range(len(divisions) - 2, -1, -1):
-        d = divisions[i]
-        if isinstance(d, float) and math.isnan(d):
+        if is_float_and_nan(divisions[i]):
             divisions[i] = divisions[i + 1]
+
+    for i in range(1, len(divisions)):
```

Comment:
This is presumably only valuable for the last few divisions, no?

Reply:
correct, I can change it to start at the end and short circuit a bit once a non-nan value is found.

Reply:
I wouldn't expect short-circuiting to be necessary.  I was more expecting this to start a few entries from the end as you suggest.